supervisor/api/deprecation.py

```diff
@@ -0,0 +1,85 @@
+"""Handle deprecation of this API."""
+import datetime
+import logging
+
+from aiohttp.web import Request, RequestHandler, Response, middleware
+
+from ..addons import Addon
+from ..const import REQUEST_FROM
+from ..coresys import CoreSys, CoreSysAttributes
+from ..homeassistant import HomeAssistant
+from ..host import HostManager
+from ..plugins import PluginObserver
+
+_LOGGER: logging.Logger = logging.getLogger(__name__)
+
+DEFAULT_GRACE_PERIOD = datetime.timedelta(seconds=10)
+
+DEPRECATED_PATHS = {
+    "/hardware/trigger": {
+        "grace_period": datetime.timedelta(hours=1),
+    }
+}
+
+
+class DeprecationMiddleware(CoreSysAttributes):
+    """Deprecation middleware functions."""
+
+    def __init__(self, coresys: CoreSys):
+        """Initialize security middleware."""
+        self.coresys: CoreSys = coresys
+        self._trottle = {}
+
+    @middleware
+    async def check_deprecation(
+        self, request: Request, handler: RequestHandler
+    ) -> Response:
+        """Check if the requested path is deprecated."""
+        if request.path in DEPRECATED_PATHS:
+            now = datetime.datetime.now()
+            if request.path not in self._trottle or (
+                now - self._trottle[request.path]
+            ) > DEPRECATED_PATHS[request.path].get(
+                "grace_period", DEFAULT_GRACE_PERIOD
+            ):
+                self._trottle[request.path] = now
```

Comment:
This should be pr. path and request_from, if you have multiple addons, it should give multiple messages